src/c_api.cpp

```diff
@@ -1594,6 +1597,69 @@ int LGBM_BoosterLoadModelFromString(
   API_END();
 }
 
+// extracts the value for key in a parameter line. E.g. for the key boosting and
+// a line like `[boosting: gbdt]` (without newline character), the extracted value would be "gbdt".
+std::string extract_param(std::string key, std::string line) {
+  std::string prefix = "[" + key + ": ";
+  std::string suffix = "]";
+  std::string prefix_replaced = line.replace(line.begin(), line.begin() + prefix.length(), "");
+  std::string suffix_replaced = prefix_replaced.replace(prefix_replaced.end() - suffix.length(), prefix_replaced.end(), "");
+  return suffix_replaced;
+}
+
+int LGBM_BoosterGetConfig(
+  BoosterHandle handle,
+  int64_t buffer_len,
+  int64_t* out_len,
+  char* out_str
+) {
+  API_BEGIN();
+  Booster* ref_booster = reinterpret_cast<Booster*>(handle);
+  std::string params = ref_booster->GetBoosting()->loaded_parameter_;
+
+  // Converts parameters to Json
+  Json::object obj = Json::object{};
+  std::stringstream input_stream(params);
+  std::string line;
+
+  if (out_str != NULL)
+  {
+    while(std::getline(input_stream, line, '\n')){
+      if (line.rfind("[boosting: ", 0) == 0) {
+        obj["boosting"] = Json{extract_param("boosting", line)};
+      } else if (line.rfind("[boost_from_average: ", 0) == 0) {
+        obj["boost_from_average"] = Json{std::stod(extract_param("boost_from_average", line)) == 1 ? true : false};
+      } else if (line.rfind("[learning_rate: ", 0) == 0) {
+        obj["learning_rate"] = Json{std::stod(extract_param("learning_rate", line))};
+      } else if (line.rfind("[metric: ", 0) == 0) {
+        obj["metric"] = Json{extract_param("metric", line)};
+      } else if (line.rfind("[monotone_constraints: ", 0) == 0) {
+        obj["monotone_constraints"] = Json{LightGBM::Common::StringToArray<int8_t>(extract_param("monotone_constraints", line), ',')};
+      } else if (line.rfind("[num_iterations: ", 0) == 0) {
+        obj["num_iterations"] = Json{std::stoi(extract_param("num_iterations", line))};
+      } else if (line.rfind("[num_leaves: ", 0) == 0) {
+        obj["num_leaves"] = Json{std::stoi(extract_param("num_leaves", line))};
+      } else if (line.rfind("[objective: ", 0) == 0) {
+        obj["objective"] = Json{extract_param("objective", line)};
+      } else if (line.rfind("[tree_learner: ", 0) == 0) {
+        obj["tree_learner"] = Json{extract_param("tree_learner", line)};
+      } else if (line.rfind("[verbose: ", 0) == 0) {
+        obj["verbose"] = Json{std::stoi(extract_param("verbose", line))};
```

Comment:
I think we can maintain a Key-value map, and load parameters in a for-loop, rather than use many if-elsees.

Reply:
Per #5244, I would find extracting the `categorical_feature` param helpful as well.

Reply:
> would find extracting the categorical_feature param helpful as well

this is now being addressed in #5424 . @johnpaulett if you're interested in influencing the shape that this support takes in LightGBM, you're welcome to leave review comments on that PR.